app/models/king.rb

```diff
@@ -1,7 +1,51 @@
 class King < Piece
+  def valid_move?(to_row, to_col)
```

Comment:
A possible alternative is to create a method just for checking if the path between the castlers is clear. This method could also make sure that the squares in the path cannot be attacked. Something like:
```
def path_is_clear?(castle_path)
  # this is mostly pseudo-code
  castle_path.each {|square| return false if square_is_empty && square_is_not_attackable}
  true
end
```